### PR TITLE
HTML-CSS-Popup task

### PR DESCRIPTION
**Pull request by lalalilulu**
### HTML-CSS-Popup — #done

repo with the task - https://github.com/lalalilulu/html-css-popup
Please find my solution on gihub pages: https://lalalilulu.github.io/html-css-popup/

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

**Comment by krylenger** [comment]
Well done!
Some things from my side:

According to homework requirements: "... IMPORTANT! Put only code required to conduct review, i.e. html, css, js files and such; images, IDE directories like .idea, .vscode etc are not required for code review; never have .git directory in your task submissions)

As far as I see, you've added 23 files, please remove them and leave just .html and .css

**Comment by krylenger** [comment]
And please add to you PR link to your original codebase and link to demo webpage 
Example:
Screenshot 2020-11-20 at 02 14 08

**Comment by lalalilulu** [comment]
Delete the icons, my repo with the task - https://github.com/lalalilulu/html-css-popup 
Please find my solution on gihub pages:   https://lalalilulu.github.io/html-css-popup/

**Comment by A-Ostrovnyy** [comment]
@lalalilulu please add links to your first message:

image

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
See my comments below.

File: submissions/lalalilulu/hooli-style_popup/index.html [hunk 1]
```diff
@@ -0,0 +1,108 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:4]: Add meta tag for mobile devices and for IE browsers.

File: submissions/lalalilulu/hooli-style_popup/index.html [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    <title>Hooli-style Popup</title>
+    <link rel="stylesheet" href="styles.css">
+</head>
+<body>
+<header>
+    <nav>
+        <ul class="menu">
+            <li>Mail</li>
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:12]: These items must be a link.

File: submissions/lalalilulu/hooli-style_popup/index.html [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                    </div>
+                </div>
+            </div>
+        </div>
+            <img class="icon" src="icons/bell.png">
+            <img  class="avatar" width="24" height="24" src="icons/avatar.png">
+    </nav>
+</header>
+</body>
+</html>
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:108]: Add an empty line to the end of each file. [Reason](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline)

File: submissions/lalalilulu/hooli-style_popup/index.html [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+                            <span>Promos</span>
+                        </div>
+                        <div class="more-last-item">
+                            <a href="https://kottans.org/" class="kottans">Even more from Kottans</a>
+                        </div>
+                    </div>
+                </div>
+            </div>
+        </div>
+            <img class="icon" src="icons/bell.png">
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:103]: These items must be a link and menu item.

File: submissions/lalalilulu/hooli-style_popup/index.html [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+        <ul class="menu">
+            <li>Mail</li>
+            <li>Images</li>
+        </ul>
+        <div class="popup">
+            <img class="popup-icon" src="icons/popup-button.png">
+            <div class="popup-container">
+                    <div class="arrow"></div>
+                    <div class="popup-items">
+                        <div class="popup-item">
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:20]: `.popup-item` must be a link. The user should be able to click on the links inside your popup.

File: submissions/lalalilulu/hooli-style_popup/index.html (hunk 2 above)
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:12]: Your popup menu not accessible from keyboard. I can't use your popup using keyboard.

File: submissions/lalalilulu/hooli-style_popup/styles.css [hunk 6]
```diff
@@ -0,0 +1,166 @@
+@import url(https://fonts.googleapis.com/css?family=Roboto:wght@300&display=swap);
+
+nav {
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/styles.css:3]: Avoid styling by id.
Separate responsibilities:
- HTML for markup
- Classes for styling
- Id for JS and for linking label to the interractive elements

File: submissions/lalalilulu/hooli-style_popup/styles.css [hunk 7]
```diff
@@ -0,0 +1,166 @@
+@import url(https://fonts.googleapis.com/css?family=Roboto:wght@300&display=swap);
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/styles.css:1]: Why you import this font if you don't use it?)

File: submissions/lalalilulu/hooli-style_popup/styles.css [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+    text-decoration: none;
+    color: #4a4d4f;
+}
+
+a:hover, a:focus {
+    color: #7a7b7d;
+    border-bottom: 1px solid #000;
+}
+
+
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/styles.css:164]: Avoid unused lines.

**Comment by lalalilulu** [comment]
The code is changed according to received comments

File: submissions/lalalilulu/hooli-style_popup/index.html [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    <title>Hooli-style Popup</title>
+    <link rel="stylesheet" href="styles.css">
+</head>
+<body>
+    <nav>
+        <ul class="menu">
+            <li><a href="#" class="nav-text">Mail</a></li>
+            <li><a href="#" class="nav-text">Images</a></li>
+            <li><div class="popup">
+                <img class="popup-icon" src="icons/popup-button.png" alt="popup">
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/index.html:16]: Remove menu elements shifting on hover, it's bad for page performance. [Watch video](https://share.vidyard.com/watch/Cmwa1XM1CmXbcxK2Xtz8ba?)

File: submissions/lalalilulu/hooli-style_popup/styles.css (hunk 6 above)
Comment (A-Ostrovnyy) [reply submissions/lalalilulu/hooli-style_popup/styles.css:3]: My bad, I forget to add HTML tag. avoid styling by HTML tag too.)

**Comment by A-Ostrovnyy** [comment]
Good work!
But I still can't open/close your popup using only keyboard. Please fix it.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
Good work!
But I still can't open/close your popup using only keyboard. Please fix it.

**Comment by lalalilulu** [comment]
code is updated -  it is possible to use only keyboard to interact with popup elements

File: submissions/lalalilulu/hooli-style_popup/styles.css [hunk 10]
```diff
@@ -30,10 +30,30 @@ nav {
     object-fit: cover;
 }
 
+.main-popup {
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/styles.css:33]: Watch the [video](https://www.youtube.com/watch?v=E6kLaaQFctU&t=413s) about hiding checkboxes in the right way.

File: submissions/lalalilulu/hooli-style_popup/styles.css [hunk 11]
```diff
@@ -68,70 +87,70 @@ a:focus {
...
 
-.popup:hover .popup-container {
-    top: 6%;
-    opacity: 1;
-}
-
-.popup-container:hover {
-    top: 6%;
-    opacity: 1;
+li > input:checked ~ .popup-container {
```
Comment (A-Ostrovnyy) [inline submissions/lalalilulu/hooli-style_popup/styles.css:105]: Avoid styling by HTML tags, use CSS classes instead.

File: submissions/lalalilulu/hooli-style_popup/index.html (hunk 9 above)
Comment (A-Ostrovnyy) [reply submissions/lalalilulu/hooli-style_popup/index.html]: Not fixed yet.

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
LGTM
Don't forget to fix items shifting on hover/focus)

File: submissions/lalalilulu/hooli-style_popup/styles.css (hunk 10 above)
Comment (lalalilulu) [reply submissions/lalalilulu/hooli-style_popup/styles.css:33]: thank you - the video helps a lot

**Comment by lalalilulu** [comment]
update files according to received  comments

**Review by A-Ostrovnyy — CHANGES_REQUESTED** [review]
I see you have a problem with shifting elements on hover.
Tip: Add transparent border to all interactive elements and change border color from transparent to color what you want on hover/focus.)

**Comment by lalalilulu** [comment]
thank you! I have added a transparent border

**Review by A-Ostrovnyy — APPROVED** [review]
Well done!

**Pull request merged**